src/main/java/com/springweb/dv_spring_web_mongo/service/ProjectService.java

```diff
@@ -30,21 +27,30 @@
     }
 
     public ProjectDTO getProjectById(String id) {
-        Optional<Project> project = projectRepository.findById(id);
-        return project.get().convertToDTO();
+        checkProjectNotNull(id);
+        return projectRepository.findById(id).get().convertToDTO();
     }
 
     public void addNewProject(ProjectDTO projectDTO) {
         projectRepository.save(projectDTO.convertToProject());
     }
 
     public void changeProjectName(ProjectDTO projectDTO, String id) {
+        checkProjectNotNull(id);
         Project project = projectRepository.findById(id).get();
         project.setProjectName(projectDTO.getProjectName());
         projectRepository.save(project);
     }
 
     public void deleteProject(String id) {
+        checkProjectNotNull(id);
         projectRepository.deleteById(id);
     }
+
+    public void checkProjectNotNull(String id) {
```

Comment:
Please return `Project` here and rename to `getById`. And you can use the method instead of repeated retrieving data in the several methods